Sources/CartingCore/Services/ProjectService.swift

```diff
@@ -264,28 +258,34 @@ public final class ProjectService {
         }
         let folder = try Folder(path: directoryPath)
         return folder.subfolders
-            .filter { folder in
-                let projectName = folder.name.deleting(suffix: "." + Constants.projectExtension)
-                var isValid = folder.name.hasSuffix(Constants.projectExtension)
-                if filterNames.isEmpty == false {
-                    isValid = isValid && filterNames.contains(projectName)
+                .filter { folder in
+                    let projectName = folder.name.deleting(suffix: "." + Constants.projectExtension)
+                    var isValid = folder.name.hasSuffix(Constants.projectExtension)
+                    if filterNames.isEmpty == false {
+                        isValid = isValid && filterNames.contains(projectName)
+                    }
+                    return isValid
                 }
-                return isValid
-            }
-            .map { $0.path }
+                .map { $0.path }
     }
 
-    private func frameworksInformation() throws -> [Framework] {
-        let frameworkFolder = try projectFolder.subfolder(atPath: "Carthage/Build/iOS")
-        let frameworks = frameworkFolder.subfolders.filter { $0.name.hasSuffix("framework") }
-        return try frameworks.map(information)
+    private func frameworksInformation() throws -> [String: [Framework]] {
+        let frameworkFolders = Dictionary(uniqueKeysWithValues: try frameworksDirectoryPaths.map { path in
+            (path, try projectFolder.subfolder(atPath: "\(path)/Build/iOS").subfolders)
```

Comment:
For now I see several places with hardcoded path components. Is it possible to combine all paths in one place like `PathsDispatcher` or something?

Reply:
I think this is good idea, will implement that